strawberry/permission.py

```diff
@@ -87,7 +100,81 @@ class AutoDirective:
 
             self._schema_directive = AutoDirective()
 
-        return self._schema_directive
+        return [self._schema_directive]
+
+    @property
+    def is_async(self) -> bool:
+        return iscoroutinefunction(self.has_permission)
+
+    def __and__(self, other: BasePermission):
+        return AndPermission(self, other)
+
+    def __or__(self, other: BasePermission):
+        return OrPermission(self, other)
+
+
+class BoolPermission(BasePermission, abc.ABC):
+    left: BasePermission
+    right: BasePermission
+
+    def __init__(self, left: BasePermission, right: BasePermission):
```

Comment:
This should accept a list of conditions to support `X & Y & Z` without nesting

Reply:
This is such an obvious improvement, cheers!

Reply:
(Halfway through this change, probably finish it off next week)